profiles/views.py

```diff
@@ -26,3 +33,13 @@ class ProfileViewSet(mixins.RetrieveModelMixin, mixins.UpdateModelMixin, viewset
     serializer_class = ProfileSerializer
     queryset = Profile.objects.all()
     lookup_field = 'user__username'
+
+
+@cache_page(60*60*24)
+def initialized_avatar(request, username, size, color, bgcolor):  # pylint:disable=unused-argument
+    """ View for initial avatar """
+    user = User.objects.filter(username=username).first()
```

Comment:
We're primarily concerned with the user's profile name, right? In that case I think we should change this query to fetch the profile name only:

```python
profile_name = Profile.objects.filter(user__username=username).values_list('name', flat=True).first()
```